numpy/core/tests/test_numeric.py

```diff
@@ -2001,10 +2003,10 @@ def test_clip_problem_cases(self, arr, amin, amax, exp):
     def test_clip_scalar_nan_propagation(self, arr, amin, amax):
         # enforcement of scalar nan propagation for comparisons
         # called through clip()
-        expected = np.minimum(np.maximum(a, amin), amax)
+        expected = np.minimum(np.maximum(arr, amin), amax)
         with assert_warns(DeprecationWarning):
             actual = np.clip(arr, amin, amax)
-            assert_equal(actual, expected)
+        assert_equal(actual, expected)
```

Comment:
Why this change? If possible, only the call that produces the warning should be in the protected block.

Reply:
That's the change. ``np.clip`` produces the ``DeprecationWarning``. This change moves the assert equal *outside* of the block

Reply:
right, sorry, I read it backwards.